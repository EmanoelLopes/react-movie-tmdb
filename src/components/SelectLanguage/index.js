import React from 'react';

const languages = [
  {value: 'none', name: '--- Languages ---'},
  {value: 'pt-BR', name: 'Português (BR)'},
  {value: 'es-ES', name: 'Español'},
  {value: 'en-US', name: 'English'},
];

const defaultLanguage = localStorage.getItem('current_language');

const SelectLanguage = () => {
  const setCurrentLanguage = event => {
    localStorage.setItem('current_language', event.target.value);
    setTimeout(() => {
      window.location.reload();
    }, 100);
  };

  return (
    <select
      defaultValue={defaultLanguage}
      onChange={setCurrentLanguage}>
      {
        languages.map((lang, index) => {
          return (
            <option
              disabled={index === 0}
              key={index}
              value={lang.value}>
              {lang.name}
            </option>
          );
        })
      }
    </select>
  );
};

export default SelectLanguage;
