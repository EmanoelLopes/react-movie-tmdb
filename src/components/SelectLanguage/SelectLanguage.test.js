import React from 'react';
import { mount } from 'enzyme';
import SelectLanguage from 'components/SelectLanguage';

let wrapped;

beforeEach(() => {
  wrapped = mount(<SelectLanguage />);
});

afterEach(() => {
  wrapped.unmount();
});

describe('<SelectLanguage />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
