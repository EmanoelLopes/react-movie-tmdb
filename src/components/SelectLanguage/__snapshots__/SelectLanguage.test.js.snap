// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectLanguage /> should match snapshot 1`] = `
<SelectLanguage>
  <select
    defaultValue={null}
    onChange={[Function]}
  >
    <option
      disabled={true}
      key="0"
      value="none"
    >
      --- Languages ---
    </option>
    <option
      disabled={false}
      key="1"
      value="pt-BR"
    >
      Português (BR)
    </option>
    <option
      disabled={false}
      key="2"
      value="es-ES"
    >
      Español
    </option>
    <option
      disabled={false}
      key="3"
      value="en-US"
    >
      English
    </option>
  </select>
</SelectLanguage>
`;
